software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java

```diff
@@ -613,22 +614,29 @@ protected ScriptHelper newScript(Map<String, ?> flags, String phase) {
                 // no pid, not running; 1 is not running
                 s.requireResultCode(Predicates.or(Predicates.equalTo(0), Predicates.equalTo(1)));
             } else if (STOPPING.equals(phase)) {
+                String stopCommand = Joiner.on('\n').join("PID=$(cat "+pidFile + ")",
+                        "test -n \"$PID\" || exit 0",
+                        "SIGTERM_USED=\"\"",
+                        "for i in $(seq 1 16); do",
+                        "  if ps -p $PID > /dev/null ; then",
+                        "     kill $PID",
+                        "     echo $PID is still running.",
+                        "  else",
+                        "     echo Process $PID stopped successfully.",
+                        "     SIGTERM_USED=\"true\"",
+                        "     break",
+                        "  fi",
+                        "  sleep 1",
+                        "done",
+                        "if test -z $SIGTERM_USED; then",
+                        "  kill -9 $PID",
+                        "  echo Process $PID stopped with SIGKILL",
```

Comment:
Same here - `Sent SIGKILL to $PID`. The process could be locked in kernel space so even `kill -9` won't kill it.